assets/js/hocs/with-transform-single-select-to-multiple-select.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore - Ignoring because @wordpress/element library does not have type definition for FunctionComponent
+// eslint-disable-next-line
+import { FunctionComponent } from '@wordpress/element';
+
+interface OriginalComponentProps {
+	selected?: number | string | null;
+}
+
+/**
+ * HOC that transforms a single select to a multiple select.
+ *
+ * @param {FunctionComponent< Record< string, unknown > >} OriginalComponent Component being wrapped.
+ */
+const withTransformSingleSelectToMultipleSelect = (
+	OriginalComponent: FunctionComponent< Record< string, unknown > >
+) => {
+	return ( props: OriginalComponentProps ): JSX.Element => {
+		let { selected } = props;
+		selected = selected || null;
```

Comment:
In the previous implementation default value `null` would be assigned only if `selected` is `undefined`, while in this case it will be assigned whenever `selected` is falsy. That also includes the cases when the value of `selected` is `0` or `''` which seem to be valid values.

Reply:
Nice catch! Just fixed it!